[JEP-305] Updated to work with Incrementals

jglick opened the pull request:
As suggested by @olivergondza, trying to fix the automated LTS branch setup so that something like https://github.com/jenkinsci/jenkins/pull/3468 does not need to be done manually the next time.

jglick committed:
[JEP-305] Updated to work with Incrementals.

jglick commented on bin/init-lts-line:
```diff
@@ -11,22 +11,22 @@ baseline=$1
 branch_name="stable-${baseline}"
-git checkout -b $branch_name jenkins-${baseline}
+# Use reincrementalification commit as branch point: second child of release commit (first is `[maven-release-plugin] prepare for next development iteration`)
+git checkout -b $branch_name $(git log --reverse --ancestry-path --pretty=%H jenkins-${baseline}..master | head -2 | tail -1)
```
The most recent example: https://github.com/jenkinsci/jenkins/commit/bc37379c96daef081c2ed0fa7ed85a06eadfaa80

jglick replied:
This needs to be amended as of https://github.com/jenkinsci/jenkins/pull/3567 to just be `| head -1` (i.e., first child of release commit).

jglick commented on bin/init-lts-line:
```diff
@@ -11,22 +11,22 @@ baseline=$1
 
 # Update versions
-mvn -q versions:set -DnewVersion=${baseline}.1-SNAPSHOT
+mvn -q versions:set-property -Dproperty=revision -DnewVersion=${baseline}.1
```
So, if all goes well,

```diff
-<revision>2.126</revision>
+<revision>2.125.1</revision>
```

jglick commented on bin/init-lts-line:
```diff
@@ -11,22 +11,22 @@ baseline=$1
+mvn -q versions:set-property -Dproperty=revision -DnewVersion=${baseline}.1
 
-# Point scm tags back to HEAD
-sed -i 's/<tag>.*<\/tag>/<tag>HEAD<\/tag>/' pom.xml */pom.xml
```
Unnecessary, because we are no longer branching from the release commit. The `<tag>` was already fixed in its first child.

jglick commented on bin/init-lts-line:
```diff
@@ -11,22 +11,22 @@ baseline=$1
 
-# Point scm tags back to HEAD
-sed -i 's/<tag>.*<\/tag>/<tag>HEAD<\/tag>/' pom.xml */pom.xml
+echo -Plts-release >> .mvn/maven.config
```
Automating what `publish-lts-rc.sh` was doing for all branch builds.

(`-Dbuild.type=rc` is useless for Incrementals and support for this was dropped in https://github.com/jenkinsci/jenkins/pull/3430.)

jglick commented on bin/publish-lts-rc.sh:
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+# TODO consider deleting in favor of binaries automatically deployed to https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/
```
And presumably some other stuff in other repositories could be deleted, like `jenkins-stable-rc.mk` wherever that is.

olivergondza commented:
Thanks Jesse. I intend to postpone merging these modifications until we have a place to run the tests. Pinging [INFRA-1724](https://issues.jenkins-ci.org/browse/INFRA-1724)...

jglick committed:
As of https://github.com/jenkinsci/jenkins/pull/3567 there should be no separate reincrementalification commit.
Brings this script into closer alignment with: https://gist.github.com/jglick/86a30894446ed38f918050c1180483e2

jglick committed:
Merge branch 'master' into incrementals

jglick commented:
> I intend to postpone merging these modifications until we have a place to run the tests.

@olivergondza I do not understand. This has nothing to do with INFRA-1724. The removal of ATH runs in https://github.com/jenkinsci/jenkins/pull/3468 was a one-off thing, because they do not pass in _that_ branch. I think _this_ PR can and should be merged prior to cutting the next LTS line.

olivergondza commented:
@jglick, got it. Thanks
